test_communication/CMakeLists.txt

```diff
@@ -24,35 +21,27 @@ ament_auto_find_build_dependencies()
 
 option(SECURITY "Activate security" OFF)
 
-set(message_files
-  "msg/BoundedArrayNested.msg"
-  "msg/BoundedArrayPrimitives.msg"
-  "msg/Builtins.msg"
-  "msg/DynamicArrayNested.msg"
-  "msg/DynamicArrayPrimitives.msg"
-  "msg/DynamicArrayPrimitivesNested.msg"
-  "msg/Empty.msg"
-  "msg/Nested.msg"
-  "msg/Primitives.msg"
-  "msg/StaticArrayNested.msg"
-  "msg/StaticArrayPrimitives.msg"
-)
-set(other_message_files
-  "msg/FieldsWithSameType.msg"
-  "msg/UInt32.msg"
-)
-set(service_files
-  "srv/Empty.srv"
-  "srv/Primitives.srv"
-)
-rosidl_generate_interfaces(${PROJECT_NAME}
-  ${message_files}
-  ${other_message_files}
-  ${service_files}
-  DEPENDENCIES builtin_interfaces
-)
-
 if(BUILD_TESTING)
+
+  find_package(ament_cmake REQUIRED)
+  find_package(test_msgs REQUIRED)
+
+  ament_index_get_resource(interface_files "rosidl_interfaces" "test_msgs")
+  string(REPLACE "\n" ";" interface_files "${interface_files}")
+
+  string(REGEX MATCHALL "([A-Z][A-Za-z0-9]*\.msg)" message_files "${interface_files}")
+  string(REGEX MATCHALL "([A-Z][A-Za-z0-9]*\.srv)" service_files "${interface_files}")
```

Comment:
Rather than embedding knowledge about the naming conventions of the interface files I would recommend to iterate over the list and append them to either `message_files` or `service_files` based on the extension  (using [string_ends_with](https://github.com/ament/ament_cmake/blob/d60d7ab800d9046f35b8c68a7d073a712cdae610/ament_cmake_core/cmake/core/string_ends_with.cmake#L25)).

Reply:
Yeah I liked this approach because I could get all matches in one invocation without having to iterate. (I started with list(FILTER REGEX) until I realized that it's only available since CMake 3.6...)

I'll change that to iterate on the list manually then

Reply:
addressed in https://github.com/ros2/system_tests/pull/223/commits/66a7db66f03e1c13962ae806a9da4fcfe9b175da